### ISSUE TITLE
Improve (wat2wasm -r) support (or remove it completely).

### ISSUE THREAD

**Issue by sbc100** [WebAssembly/wabt#1137]
It seems to cause some confusion because it doesn't, for example, write out a symbols table.

See:
https://stackoverflow.com/questions/57294541/error-linking-wasm-object-files-with-wasm-ld/57367477#57367477

We could improve it but I'm not who this would benefit exactly.  Do we have users for such a feature?

**Comment by binji** [comment]
Yes, I think that's the same thing we were discussing here: https://github.com/WebAssembly/wabt/issues/1133

**Comment by srenatus** [comment]
I might be mistaken, but to work with externref (reference types) from a language like C/C++, where there's no such support in LLVM in sight (correct me if I'm wrong please), the following seems like a way to go:

* create a wat file with some simple functions using externref,
* `wat2wasm -r` it,
* use lld (`wasm-ld`) to link it with the wasm object files that clang has compiled the C/C++ sources to.

🤔 What approach to this do other folks use? 😃

**Comment by sbc100** [comment]
Actually there are plans underway to add externref support for llvm at all layers up to included C/C++.    See https://github.com/Igalia/ref-cpp.    There are even several patches that have landed and several under review right now (See https://reviews.llvm.org/D95425 for example).

The doesn't mean we would not also welcome improvements to `wat2wasm -r`

**Comment by srenatus** [comment]
@sbc100 thanks those pointers are useful!

I must admit I'm having difficulties navigating reviews.llvm.org. Has any of this been in llvm-12 already, would you know? 🔍

**Comment by sbc100** [comment]
I would recommend using tot (tip-of-tree) if you want to try any of these features.  I'm not sure anything made it into llvm-12.  I'm also not sure what has landed even on tot.   If you want to use this stuff you will be right on bleeding edge.

**Comment by srenatus** [comment]
@sbc100 thanks again 😃

**Comment by keithw** [comment]
Closing as `wat2wasm --relocatable` support seems to have been improved in #1537, #1549, etc. The example in the linked Stack Overflow post now works fine.
